### PR TITLE
Added login view to project

### PR DESCRIPTION
**Pull request by heliomar-pena**
### Explanation

Hello everyone! I've added a basic logic to log user, just do the query and receive the user from database. I know this will change after, but I want to do it more easy when we've all that we need.

I've tested a lot of things trying to get the best way to do the things using Vue 3 and Vue apollo.

I need to test some things to maintain the user session opened, for this test i have used localStorage but @Israel-Laguan suggest to me to try a library that I want to test after

- [Added quasar notify to project](https://github.com/Azordev/did-app/commit/c66b3ab1260dc7d708726096481746970e89a22c)
- [linting tsconfig.json](https://github.com/Azordev/did-app/commit/8bed74295e2be3cb0d326f5ff9c08ce2cb4d4a1b)
- [Fixing lint error about import .ts files](https://github.com/Azordev/did-app/commit/892b0723e70ac12730a43b5a14b1d18ff960931e)
- [Added actions folders and user actions](https://github.com/Azordev/did-app/commit/8e23c381207bad3bb9fede27cbbc583041b8f058)
- [Added validations file in utils](https://github.com/Azordev/did-app/commit/3bd696e6cfa5f2736fe17fdab6a48c8ceb1e29ca)
- [Created login logic](https://github.com/Azordev/did-app/commit/e190ab0b006b7d95eebba6e03f9a5b02b0cf0097)
- [Adding LocalStorage plugin to quasar](https://github.com/Azordev/did-app/commit/bb1002fb0a230372ed6ec03187d1cae27b0734c9)
- [Improving the user login logic](https://github.com/Azordev/did-app/commit/95539ea72891df6318b4fcbb083b683196cdf7c1)

### Screenshots
lastscreenshot

Button loading
lastscreenshot

Password and email validations
lastscreenshot

Error Message
lastscreenshot

Success Message
lastscreenshot

### CODE REVIEW

File: quasar.config.js [hunk 1]
```diff
@@ -185,32 +182,28 @@ module.exports = configure(function (/* ctx */) {
...
         // Windows only
         // win32metadata: { ... }
       },
 
       builder: {
         // https://www.electron.build/configuration/configuration
 
-        appId: 'diprap'
-      }
+        appId: 'diprap',
```
Comment (Israel-Laguan) [inline quasar.config.js]: please change name here

File: quasar.config.js [hunk 2]
```diff
@@ -80,20 +74,23 @@ module.exports = configure(function (/* ctx */) {
...
-        }]
-      ]
+        [
+          '@intlify/vite-plugin-vue-i18n',
+          {
+            // if you want to use Vue I18n Legacy API, you need to set `compositionOnly: false`
+            // compositionOnly: false,
+
+            // you need to set i18n resource including paths !
+            include: path.resolve(__dirname, './src/i18n/**'),
```
Comment (Israel-Laguan) [inline quasar.config.js:86]: maybe not needed, but is ok for me to keep just in case.

File: src/pages/Login/Login.css [hunk 3]
```diff
@@ -0,0 +1,5 @@
+.login {
+  padding: 40px 20px 0;
```
Comment (Israel-Laguan) [inline src/pages/Login/Login.css:2]: use variables! ... wait, sorry, is the Sankeerna trauma, disregard this comment hehe

File: src/utils/apollo.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  useQuery as _useQuery,
+} from '@vue/apollo-composable';
+
+import { DocumentNode } from 'graphql';
+
+import { Notify } from 'quasar';
+
+const httpLink = createHttpLink({
+  // You should use an absolute URL here
+  uri: 'https://did-db.herokuapp.com/v1/graphql',
```
Comment (Israel-Laguan) [inline src/utils/apollo.ts]: optional: move to a ENV variable

File: src/utils/validations.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  (val: string) =>
+    val.match(/(?=.*?[a-z])/) ||
+    'Password should have min one lower case English Letter',
+  (val: string) =>
+    val.match(/(?=.*?[0-9])/) || 'Password should have min one number',
+  (val: string) =>
+    val.match(/(?=.*?[#?!@$%^&*-])/) ||
+    'Password should have min one special character',
+  (val: string) =>
+    val.match(/.{8,}/) || 'Password should have min eight letters',
```
Comment (Israel-Laguan) [inline src/utils/validations.ts]: Please propose these validations to Daniel so he approve them

File: src/utils/apollo.ts (hunk 4 above)
Comment (heliomar-pena) [reply src/utils/apollo.ts]: Of course! Will do. Thank you

File: src/utils/validations.ts (hunk 5 above)
Comment (heliomar-pena) [reply src/utils/validations.ts]: They're examples to apply the correct rules when we've it, but could be a good moment to fix that details, let me asking he

**Review by e-azocar — APPROVED** [review]
It's ok, but, I have a comment.

Isn't the symbol `.` a special character?
image

**Comment by heliomar-pena** [comment]
> It's ok, but, I have a comment.
> 
> Isn't the symbol `.` a special character? image

Yes, it should be. I'll add it

File: src/utils/validations.ts (hunk 5 above)
Comment (heliomar-pena) [reply src/utils/validations.ts]: I've does the question on the project group

**Comment by heliomar-pena** [comment]
Thanks for checking guys!. I've made changes, I will merge to work with products view now

**Pull request merged**